### PR TITLE
Non static ap ifor single pulse extractor

### PR DESCRIPTION
**Pull request by jebuss**
new API without static and global members for SPE extractor

### CODE REVIEW

File: src/main/java/fact/extraction/SinglePulseExtraction.java [hunk 1]
```diff
@@ -8,7 +8,8 @@
 import stream.annotations.Parameter;
 import java.util.Arrays;
 import java.util.ArrayList;
-import fact.features.singlePulse.timeLineExtraction.SinglePulseExtractor;
+import fact.features.singlePulse.timeSeriesExtraction.SinglePulseExtractor;
+import fact.features.singlePulse.timeSeriesExtraction.ElementWise;
```
Comment (jebuss) [inline src/main/java/fact/extraction/SinglePulseExtraction.java]: maybe we want move`fact.features.singlePulse.timeSeriesExtraction.ElementWise` to `fact.Utils` or a library folder with helper classes that we put there.

File: src/main/java/fact/extraction/SinglePulseExtraction.java [hunk 2]
```diff
@@ -57,47 +58,47 @@ public Data process(Data input) {
...
 
-
-
         addStartSliceOffset(pixelArrivalSlices);
-        // printArrivalSlices(pixelArrivalSlices);
+
         input.put(outputKey, pixelArrivalSlices);
-        input.put(outputKey+"TL", reducedTimeline);
-        input.put(outputKey+"Count", single_pe_count);
+        input.put(outputKey+"TimeSeriesAfterExtraction", flatten(timeSeriesAfterExtraction));
```
Comment (jebuss) [inline src/main/java/fact/extraction/SinglePulseExtraction.java]: we could start to use the naming convention from the fact_tools refactoring branch e.g. outputkey+":TimeSeriesAfterExtraction".

File: src/main/java/fact/extraction/SinglePulseExtraction.java [hunk 3]
```diff
@@ -110,6 +111,25 @@ private void addStartSliceOffset(int[][] arr) {
         }
     }
 
+    private double[] flatten(double[][] matrix2d) {
```
Comment (jebuss) [inline src/main/java/fact/extraction/SinglePulseExtraction.java]: lets move this to `fact.Utils`

File: src/main/java/fact/utils/ConvertSinglePulses2Timeseries.java [hunk 4]
```diff
@@ -17,41 +17,57 @@
...
-                AddFirstArrayToSecondArray.at(pulseTemplate, current_timeseries, singlePulses[pix][pulse]);
+                AddFirstArrayToSecondArray.at(
+                    pulseTemplate, 
+                    current_timeseries, 
+                    singlePulses[pix][pulse]);
             }
 
-            timeseries = (double[]) ArrayUtils.addAll(timeseries, current_timeseries);
+            // Add the baseline to the time series
+            current_timeseries = ElementWise.add(current_timeseries, baseLine[pix]);
```
Comment (jebuss) [inline src/main/java/fact/utils/ConvertSinglePulses2Timeseries.java]: naming inconsistancy

File: src/main/java/fact/extraction/SinglePulseExtraction.java (hunk 3 above)
Comment (jebuss) [inline src/main/java/fact/extraction/SinglePulseExtraction.java]: And a unit test would be tremendous. Please, make flatten 2d matrix great again!!!

**Comment by relleums** [comment]
You guys are splendid! Thank you so much!

**Pull request merged**